src/backend/distributed/commands/table.c

```diff
@@ -3403,3 +3405,44 @@ MakeNameListFromRangeVar(const RangeVar *rel)
 		return list_make1(makeString(rel->relname));
 	}
 }
+
+
+/*
+ * WarnUnsupportedIfForeignDistributedTable gets a relationId and logs a WARNING
+ * if the given relation is a distributed foreign table.
+ * We do that because now we only support Citus Local Tables for foreign tables.
+ */
+void
+WarnUnsupportedIfForeignDistributedTable(Oid relationId)
+{
+	if (!DistributedForeignTableWarningPrompted && IsForeignTable(relationId) &&
+		IsCitusTable(relationId) && !IsCitusTableType(relationId, CITUS_LOCAL_TABLE))
+	{
+		DistributedForeignTableWarningPrompted = true;
+		ereport(WARNING, (errmsg("support for distributed foreign tables are deprecated, "
+								 "please use Citus managed local tables"),
+						  (errdetail("Foreign tables can be added to metadata using UDF: "
+									 "citus_add_local_table_to_metadata()"))));
+	}
+}
+
+
+/*
+ * WarnIfListHasForeignDistributedTable iterates the given list and calls
+ * WarnUnsupportedIfForeignDistributedTable for each relation, which logs
+ * a WARNING message if it's a distributed foreign table.
+ */
+void
+WarnIfListHasForeignDistributedTable(List *rangeTableList)
+{
+	RangeTblEntry *rangeTableEntry = NULL;
+	foreach_ptr(rangeTableEntry, rangeTableList)
+	{
+		if (DistributedForeignTableWarningPrompted)
```

Comment:
why not this check is done before the loop? 
Using `DistributedForeignTableWarningPrompted` in two different functions make it pretty hard to follow. Can you make sure we only use it once? 

And, can't we define that as a function level static variable?

Reply:
See function scoped static variables: https://www.geeksforgeeks.org/static-variables-in-c/